### ISSUE TITLE
Need to list minimum required tutor version

### ISSUE THREAD

**Issue by throwaway-a** [hastexo/tutor-contrib-backup#32]
I wanted to duplicate a site running tutor 12.0.4 / Lilac. But when I went to install the plugin it upgraded me to tutor 13.3.1. That seems like a very bad thing to happen automatically, as it will break all sorts of stuff. So please list the version requirements on the site (apparently currently tutor >= 13.2.0, in which case I would have known I couldn't use this plugin.) This also somehow reactivated a vestigial Cairn plugin

```
ubuntu@ip-172-31-94-95:~$ tutor --version
tutor, version 12.0.4
ubuntu@ip-172-31-94-95:~$ sudo su tutor
tutor@ip-172-31-94-95:/home/ubuntu$ pip install git+https://github.com/hastexo/tutor-contrib-backup@v0.1.0
Collecting git+https://github.com/hastexo/tutor-contrib-backup@v0.1.0
  Cloning https://github.com/hastexo/tutor-contrib-backup (to revision v0.1.0) to /tmp/pip-req-build-69wule_a
  Running command git clone -q https://github.com/hastexo/tutor-contrib-backup /tmp/pip-req-build-69wule_a
  Running command git checkout -q 3d3ea6fc13b69338d15b74adc30dfb8d14f34cd5
Collecting tutor<14,>=13.2.0
  Downloading tutor-13.3.1.tar.gz (75 kB)
     |████████████████████████████████| 75 kB 5.0 MB/s 
Requirement already satisfied: appdirs in /home/tutor/.local/lib/python3.8/site-packages (from tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (1.4.4)
Requirement already satisfied: click>=7.0 in /usr/lib/python3/dist-packages (from tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (7.0)
Requirement already satisfied: jinja2 in /usr/lib/python3/dist-packages (from tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (2.10.1)
Requirement already satisfied: kubernetes in /home/tutor/.local/lib/python3.8/site-packages (from tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (12.0.1)
Requirement already satisfied: mypy in /home/tutor/.local/lib/python3.8/site-packages (from tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (0.910)
Requirement already satisfied: pycryptodome in /home/tutor/.local/lib/python3.8/site-packages (from tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (3.10.1)
Requirement already satisfied: pyyaml>=4.2b1 in /usr/lib/python3/dist-packages (from tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (5.3.1)
Requirement already satisfied: requests in /usr/lib/python3/dist-packages (from kubernetes->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (2.22.0)
Requirement already satisfied: urllib3>=1.24.2 in /usr/lib/python3/dist-packages (from kubernetes->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (1.25.8)
Requirement already satisfied: google-auth>=1.0.1 in /home/tutor/.local/lib/python3.8/site-packages (from kubernetes->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (1.27.1)
Requirement already satisfied: python-dateutil>=2.5.3 in /home/tutor/.local/lib/python3.8/site-packages (from kubernetes->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (2.8.1)
Requirement already satisfied: six>=1.9.0 in /usr/lib/python3/dist-packages (from kubernetes->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (1.14.0)
Requirement already satisfied: certifi>=14.05.14 in /usr/lib/python3/dist-packages (from kubernetes->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (2019.11.28)
Requirement already satisfied: websocket-client!=0.40.0,!=0.41.*,!=0.42.*,>=0.32.0 in /home/tutor/.local/lib/python3.8/site-packages (from kubernetes->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (0.58.0)
Requirement already satisfied: setuptools>=21.0.0 in /usr/lib/python3/dist-packages (from kubernetes->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (45.2.0)
Requirement already satisfied: requests-oauthlib in /home/tutor/.local/lib/python3.8/site-packages (from kubernetes->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (1.3.0)
Requirement already satisfied: typing-extensions>=3.7.4 in /home/tutor/.local/lib/python3.8/site-packages (from mypy->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (3.10.0.0)
Requirement already satisfied: mypy-extensions<0.5.0,>=0.4.3 in /home/tutor/.local/lib/python3.8/site-packages (from mypy->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (0.4.3)
Requirement already satisfied: toml in /home/tutor/.local/lib/python3.8/site-packages (from mypy->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (0.10.2)
Requirement already satisfied: rsa<5,>=3.1.4; python_version >= "3.6" in /home/tutor/.local/lib/python3.8/site-packages (from google-auth>=1.0.1->kubernetes->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (4.7.2)
Requirement already satisfied: cachetools<5.0,>=2.0.0 in /home/tutor/.local/lib/python3.8/site-packages (from google-auth>=1.0.1->kubernetes->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (4.2.1)
Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/lib/python3/dist-packages (from google-auth>=1.0.1->kubernetes->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (0.2.1)
Requirement already satisfied: oauthlib>=3.0.0 in /usr/lib/python3/dist-packages (from requests-oauthlib->kubernetes->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (3.1.0)
Requirement already satisfied: pyasn1>=0.1.3 in /usr/lib/python3/dist-packages (from rsa<5,>=3.1.4; python_version >= "3.6"->google-auth>=1.0.1->kubernetes->tutor<14,>=13.2.0->tutor-contrib-backup==0.1.0) (0.4.2)
Building wheels for collected packages: tutor-contrib-backup, tutor
  Building wheel for tutor-contrib-backup (setup.py) ... done
  Created wheel for tutor-contrib-backup: filename=tutor_contrib_backup-0.1.0-py3-none-any.whl size=15813 sha256=ee68ef932805c88fa7364acc9285807cd62ba096da6367907df3dd7abee39453
  Stored in directory: /tmp/pip-ephem-wheel-cache-bero_sgi/wheels/95/12/5f/4939537c5fe3aa4bc00211776e7c2ecdf550ef1a6dc03e5845
  Building wheel for tutor (setup.py) ... done
  Created wheel for tutor: filename=tutor-13.3.1-py3-none-any.whl size=106377 sha256=5f93bbdd23fe93f712748bcac23c7010134a3ff78207afc73a91573aeadf2b3b
  Stored in directory: /home/tutor/.cache/pip/wheels/39/cd/42/70272cdb7d98c305d81dd6432f795f601b017d6748722415de
Successfully built tutor-contrib-backup tutor
ERROR: tutor-cairn 12.0.8 has requirement tutor<13.0.0,>=12.0.0, but you'll have tutor 13.3.1 which is incompatible.
Installing collected packages: tutor, tutor-contrib-backup
  Attempting uninstall: tutor
    Found existing installation: tutor 12.0.4
    Uninstalling tutor-12.0.4:
      Successfully uninstalled tutor-12.0.4
Successfully installed tutor-13.3.1 tutor-contrib-backup-0.1.0
tutor@ip-172-31-94-95:~$ tutor --version
tutor, version 13.3.1
```

**Comment by fghaas** [comment]
If you need Tutor 12, you need to have `tutor<13` somewhere in your _own_ requirements list (however you express that), and then pip should be intelligent enough to install the prior version of the backup plugin, if both are installed in the same go.

I say "should" because we have run into pip resolver issues elsewhere (see hastexo/hastexo-xblock#216 if you're curious) where pip failed to do the right thing. But that's not something that the plugin can do anything about — as far as it is concerned, it is expressing its dependencies correctly. The latest version can only run with 13.2.0 or later, because that's when Tutor introduced the v2 plugin API, and as of version 0.1.0 this plugin is using that API.

That said, we never meant to support Tutor versions *earlier* than 13; we should probably have had *that* dependency in from the get-go, but unfortunately it's too late to retroactively do that now.

For what it's worth, this is what happens if you install with the latest pip at the time of writing (pip-22.1.2):

```
$ pip install "tutor<13" git+https://github.com/hastexo/tutor-contrib-backup@v0.1.0
Collecting git+https://github.com/hastexo/tutor-contrib-backup@v0.1.0
  Cloning https://github.com/hastexo/tutor-contrib-backup (to revision v0.1.0) to /tmp/pip-req-build-4367krue
  Running command git clone --filter=blob:none --quiet https://github.com/hastexo/tutor-contrib-backup /tmp/pip-req-build-4367krue
  Running command git checkout -q 3d3ea6fc13b69338d15b74adc30dfb8d14f34cd5
  Resolved https://github.com/hastexo/tutor-contrib-backup to commit 3d3ea6fc13b69338d15b74adc30dfb8d14f34cd5
  Preparing metadata (setup.py) ... done
Collecting tutor<13
  Using cached tutor-12.2.0-py3-none-any.whl
Collecting jinja2
  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)
Collecting appdirs
  Using cached appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
Collecting kubernetes
  Using cached kubernetes-24.2.0-py2.py3-none-any.whl (1.5 MB)
Collecting click>=7.0
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting pycryptodome
  Using cached pycryptodome-3.15.0-cp35-abi3-manylinux2010_x86_64.whl (2.3 MB)
Collecting pyyaml>=4.2b1
  Using cached PyYAML-6.0-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (701 kB)
Collecting mypy
  Using cached mypy-0.961-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (17.8 MB)
INFO: pip is looking at multiple versions of <Python from Requires-Python> to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of tutor to determine which version is compatible with other requirements. This could take a while.
Collecting tutor<13
  Using cached tutor-12.1.7-py3-none-any.whl
  Using cached tutor-12.1.6-py3-none-any.whl
  Using cached tutor-12.1.5-py3-none-any.whl
  Using cached tutor-12.1.4-py3-none-any.whl
  Downloading tutor-12.1.3.tar.gz (55 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.9/55.9 kB 460.0 kB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading tutor-12.1.2.tar.gz (55 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.8/55.8 kB 1.8 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading tutor-12.1.1.tar.gz (55 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.7/55.7 kB 2.1 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
INFO: pip is looking at multiple versions of <Python from Requires-Python> to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of tutor to determine which version is compatible with other requirements. This could take a while.
  Downloading tutor-12.1.0.tar.gz (55 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.7/55.7 kB 3.2 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading tutor-12.0.4.tar.gz (56 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 56.2/56.2 kB 1.8 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading tutor-12.0.3.tar.gz (56 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 56.1/56.1 kB 3.1 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading tutor-12.0.2.tar.gz (55 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.6/55.6 kB 2.4 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading tutor-12.0.1.tar.gz (56 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 56.3/56.3 kB 2.9 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.
ERROR: Cannot install tutor-contrib-backup==0.1.0 and tutor<13 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested tutor<13
    tutor-contrib-backup 0.1.0 depends on tutor<14 and >=13.2.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflict
```

In other words, with a current pip, nothing would have broken and pip would instead have prompted you to resolve the situation manually. If you're saddled with an ancient pip, it sadly only warns you about the version conflict but clobbers your installation anyway.
